integration/clean/clean_dedicated_host.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"log"
+	"time"
+)
+
+// Can't release a host if it was being used within the last 24 hr add 2 hr as a buffer
+var expirationDateDedicatedHost = time.Now().UTC().Add(keepDurationDedicatedHost)
+
+const keepDurationDedicatedHost = -1 * time.Hour * 26
+
+func cleanDedicatedHost() []error {
+	log.Print("Begin to clean EC2 Dedicated Host")
+
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		return []error{err}
+	}
+	ec2client := ec2.NewFromConfig(defaultConfig)
+
+	// Get list of dedicated host
+	nameFilter := types.Filter{Name: aws.String("tag:Name"), Values: []string{
+		"IntegrationTestMacDedicatedHost",
+	}}
+
+	dedicatedHosts, errors := getDedicatedHost(cxt, ec2client, nameFilter)
+	if err != nil {
+		return errors
+	}
+
+	dedicatedHostIds := make([]string, 0)
+	for _, dedicatedHost := range dedicatedHosts {
+		log.Printf("dedicated host id %v experation date %v dedicated host creation date raw %v",
+			*dedicatedHost.HostId, expirationDateDedicatedHost, *dedicatedHost.AllocationTime)
+		if expirationDateDedicatedHost.After(*dedicatedHost.AllocationTime) && dedicatedHost.State == types.AllocationStateAvailable {
+			log.Printf("Try to delete dedicated host %s tags %v launch-date %s", *dedicatedHost.HostId, dedicatedHost.Tags, *dedicatedHost.AllocationTime)
+			dedicatedHostIds = append(dedicatedHostIds, *dedicatedHost.HostId)
+		}
+	}
+
+	releaseDedicatedHost := ec2.ReleaseHostsInput{HostIds: dedicatedHostIds}
+	_, err = ec2client.ReleaseHosts(cxt, &releaseDedicatedHost)
+	if err != nil {
+		return []error{err}
+	}
+
+	return nil
+}
+
+func getDedicatedHost(cxt context.Context, ec2client *ec2.Client, nameFilter types.Filter) ([]types.Host, []error) {
```

Comment:
Typically I'm okay with defining small functions like this, but it feels inconsistent. [Lines 54-58](https://github.com/aws/amazon-cloudwatch-agent/pull/448/files#diff-3abfb2bd5ed887a356d3af5724ab99d1980fdd4eff3bd3fe8bc82324dafec894R54-R58) do basically the same thing this function does with a different EC2 API, but it's in the main function. In this case, I don't have a preference as to whether we functionalize everything or just dump it all in the main function, as long as it's mostly consistent.